### PR TITLE
allow direct invocation from src directory

### PR DESCRIPTION
**Pull request by cs01**
After looking at pip's instructions (https://pip.pypa.io/en/stable/development/getting-started/) I thought pipx could benefit from a similar contributor workflow. 

With this change, pipx can be run from source with `python src/pipx --version` with no setup, and no ambiguity about what is being run. The contributing instructions were updated to reflect this.

### CODE REVIEW

**Comment by itsayellow** [comment]
I'm looking at "Files changed", and I'm wondering why it doesn't show the docs/installation.md commits as being already in master?  They show up as new in this PR.

File: docs/contributing.md [hunk 1]
```diff
@@ -119,14 +100,16 @@ nox -s watch_docs
 nox -s publish_docs
 ```
 
-## Release New `pipx` Version
+## Releasing New `pipx` Versions
 To create a new release
 
+* update pipx's version in `__main__.py` and regenerate documentation
```
Comment (itsayellow) [inline docs/contributing.md]: Where is the version specified in `__main__.py`?  And why not `setup.py`?

**Comment by cs01** [comment]
Not sure what the cause of that was, but it went away after pulling from origin/master.

File: docs/contributing.md [hunk 2]
```diff
@@ -1,46 +1,27 @@
...
-```
-
-A virtual environment with required dependencies is now sandboxed and activated!
-
-Go ahead and make your changes now. You can confirm they still work by running unit tests (see the next section).
-
-To leave the environment, type `deactivate`.
-
-## Testing pipx locally
 Tests are defined as `nox` sessions. You can see all nox sessions with
```
Comment (itsayellow) [inline docs/contributing.md]: Why did you delete the section describing the develop nox sessions?

File: docs/installation.md [hunk 3]
```diff
@@ -13,15 +13,22 @@ python3 -m pip install --user pipx
...
+
 ## Upgrade pipx
 ```
 python3 -m pip install -U pipx
 ```
 
-### Installation Options
-pipx's default binary location is `~/.local/bin`. This can be overriden with the environment variable `PIPX_BIN_DIR`.
+### Note: Upgrading pipx from a pre-0.15.0.0 version to 0.15.0.0 or later
+When upgrading to pipx 0.15.0.0 or above from a pre-0.15.0.0 version, you must re-install all packages to take advantage of the new persistent pipx metadata files introduced in the 0.15.0.0 release. These metadata files store pip specification values, injected packages, any custom pip arguments, and more in each main package's venv.
```
[the remark this answers is not in the document]
Comment (itsayellow) [reply docs/installation.md]: This comment got placed as outdated because docs/installation.md is now not marked as changed, but I think it's changing this would still be good, either here or with another PR.

File: docs/contributing.md (hunk 1 above)
Comment (itsayellow) [reply docs/contributing.md]: Oh I think maybe you mean `main.py` ?

Comment (itsayellow) [reply docs/contributing.md]: And I see the fact that `setup.py` no longer has a hard-coded version.

File: docs/contributing.md (hunk 2 above)
Comment (cs01) [reply docs/contributing.md]: Because the code can be tested directly without creating a virtual environment (with exceptions around dependent packages userpath and argcomplete, though I assume those are installed since I assume the user has already installed pipx).

File: docs/contributing.md (hunk 1 above)
Comment (cs01) [reply docs/contributing.md]: Yup I meant `main.py`, thanks.

**Comment by itsayellow** [comment]
I like it.  I think this will make it easier for the maintainers to test bugs (i.e. switch between the current dev and the release version that a bug report is referring to.)
